internal/conf/configuration.go

```diff
@@ -18,6 +19,26 @@ const defaultMinPasswordLength int = 6
 const defaultChallengeExpiryDuration float64 = 300
 const defaultFlowStateExpiryDuration time.Duration = 300 * time.Second
 
+// Time is used to represent timestamps in the configuration, as envconfig has
+// trouble parsing empty strings, due to time.Time.UnmarshalText().
+type Time struct {
+	time.Time
+}
+
+func (t *Time) UnmarshalText(text []byte) error {
+	trimed := bytes.TrimSpace(text)
```

Comment:
/nit happy path is alr covered by `test.env` having `GOTRUE_SMS_TEST_OTP_VALID_UNTIL=""` but can consider writing a unit test for the function as a whole (incl. `len(trimmed) >=1`)